code/modules/modular_computers/computers/modular_computer/core.dm

```diff
@@ -0,0 +1,312 @@
+		to_chat(user, "You emag \the [src]. It's screen briefly shows a \"OVERRIDE ACCEPTED: New software downloads available.\" message.")
+		return 1
+
+/obj/item/modular_computer/update_icon()
+	icon_state = icon_state_unpowered
+
+	overlays.Cut()
+	if(bsod)
+		overlays.Add("bsod")
+		return
+	if(!enabled)
+		if(icon_state_screensaver)
+			overlays.Add(icon_state_screensaver)
+		set_light(0)
+		return
+	set_light(0.2, 0.1, light_strength)
+	if(active_program)
+		overlays.Add(active_program.program_icon_state ? active_program.program_icon_state : icon_state_menu)
+		if(active_program.program_key_state)
+			overlays.Add(active_program.program_key_state)
+	else
+		overlays.Add(icon_state_menu)
+
+/obj/item/modular_computer/proc/turn_on(var/mob/user)
+	if(bsod)
+		return
+	if(tesla_link)
+		tesla_link.enabled = 1
+	var/issynth = issilicon(user) // Robots and AIs get different activation messages.
+	if(damage > broken_damage)
+		if(issynth)
+			to_chat(user, "You send an activation signal to \the [src], but it responds with an error code. It must be damaged.")
+		else
+			to_chat(user, "You press the power button, but the computer fails to boot up, displaying variety of errors before shutting down again.")
+		return
+	if(processor_unit && (apc_power(0) || battery_power(0))) // Battery-run and charged or non-battery but powered by APC.
+		if(issynth)
+			to_chat(user, "You send an activation signal to \the [src], turning it on")
+		else
+			to_chat(user, "You press the power button and start up \the [src]")
+		enable_computer(user)
+
+	else // Unpowered
+		if(issynth)
+			to_chat(user, "You send an activation signal to \the [src] but it does not respond")
+		else
+			to_chat(user, "You press the power button but \the [src] does not respond")
+
+// Relays kill program request to currently active program. Use this to quit current program.
+/obj/item/modular_computer/proc/kill_program(var/forced = 0)
+	if(active_program)
+		active_program.kill_program(forced)
+		active_program = null
+	var/mob/user = usr
+	if(user && istype(user))
+		ui_interact(user) // Re-open the UI on this computer. It should show the main screen now.
+	update_icon()
+
+// Returns 0 for No Signal, 1 for Low Signal and 2 for Good Signal. 3 is for wired connection (always-on)
+/obj/item/modular_computer/proc/get_ntnet_status(var/specific_action = 0)
+	if(network_card)
+		return network_card.get_signal(specific_action)
+	else
+		return 0
+
+/obj/item/modular_computer/proc/add_log(var/text)
+	if(!get_ntnet_status())
+		return 0
+	return ntnet_global.add_log(text, network_card)
+
+/obj/item/modular_computer/proc/shutdown_computer(var/loud = 1)
+	kill_program(1)
+	QDEL_NULL_LIST(terminals)
+	for(var/datum/computer_file/program/P in idle_threads)
+		P.kill_program(1)
+		idle_threads.Remove(P)
+	if(loud)
+		visible_message("\The [src] shuts down.", range = 1)
+	enabled = 0
+	update_icon()
+
+/obj/item/modular_computer/proc/enable_computer(var/mob/user = null)
+	enabled = 1
```

Comment:
Boolean variables and return values should use TRUE and FALSE constans instead of 1 and 0.
[CONTRIBUTING.md](https://github.com/discordia-space/CEV-Eris/blob/master/CONTRIBUTING.md)

Reply:
fixed